datahub-web-react/src/app/search/SearchBar.tsx

```diff
@@ -179,16 +179,19 @@ export const SearchBar = ({
             disabled: true,
         };
 
-        return [...moduleOptions, exploreAllOption];
-    }, [data?.listRecommendations?.modules, onClickExploreAll]);
+        const tail = showQuickFilters ? [exploreAllOption] : [];
+
+        return [...moduleOptions, ...tail];
+    }, [data?.listRecommendations?.modules, onClickExploreAll, showQuickFilters]);
 
     const { quickFilters, selectedQuickFilter, setSelectedQuickFilter } = useQuickFiltersContext();
 
     const autoCompleteQueryOptions = useMemo(() => {
         const query = suggestions.length ? effectiveQuery : '';
-        const selectedQuickFilterLabel = selectedQuickFilter
-            ? getQuickFilterDetails(selectedQuickFilter, entityRegistry).label
-            : '';
+        const selectedQuickFilterLabel =
+            showQuickFilters && selectedQuickFilter
```

Comment:
@chriscollins3456 could I get your eyes on this check? It's not strictly needed for this PR but I felt like it was more appropriate for only running quickfilters-related logic in this component when the `showQuickFilters` was enabled. I could remove it but felt nice to throw it in here.

Reply:
I feel pretty confident this is correct, the previous version of this code didn't even show quick filters so I don't think this would break anything.

Reply:
yup this is a safe change!